src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs

```diff
@@ -62,5 +156,33 @@ private static UIElement CreateElement(string text, double lineHeight, TextForma
             border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
             return border;
         }
+
+        /// <summary>
+        /// Determines if the border is being moused over and shows the info accordingly
+        /// </summary>
+        private void Border_ToolTipOpening(object sender, ToolTipEventArgs e)
+        {
+            var border = (Border)sender;
+            e.Handled = true;
+
+            bool KeepOpen()
+            {
+                var mousePoint = Mouse.GetPosition(border);
+                return !(mousePoint.X > border.ActualWidth || mousePoint.X < 0 || mousePoint.Y > border.ActualHeight || mousePoint.Y < 0);
+            }
+
+            var toolTipPresenter = _toolTipService.CreatePresenter(_textView, new ToolTipParameters(true, false, KeepOpen));
+            var startToolTip = StartToolTipServiceAsync(toolTipPresenter);
+        }
+
+        /// <summary>
+        /// Waits for the description to be created and updates the tooltip with the associated information
+        /// </summary>
+        private async Task<IReadOnlyCollection<object>> StartToolTipServiceAsync(IToolTipPresenter toolTipPresenter)
+        {
+            var uiList = await CreateDescriptionAsync(CancellationToken.None).ConfigureAwait(false);
+            toolTipPresenter.StartOrUpdate(_textView.TextSnapshot.CreateTrackingSpan(_span.Start, _span.Length, SpanTrackingMode.EdgeInclusive), uiList);
```

Comment:
Did we get a confirmation that this is safe to use off the UI thread? Or does this need to switch back?